sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -316,27 +313,41 @@ def sdc_pandas_dataframe_append_impl(df, other, ignore_index=True, verify_integr
             column_list.append((f'new_col_{col_id}_other', col_name))
 
     data = ', '.join(f'"{column_name}": {column}' for column, column_name in column_list)
-    # TODO: Handle index
-    func_text.append(f"return pandas.DataFrame({{{data}}})\n")
+
+    if ignore_index_value == True:  # noqa
+        func_text.append(f'return pandas.DataFrame({{{data}}})\n')
+    else:
+        func_text.append(f'df_index = df.index')
+        func_text.append(f'other_index = other.index')
+        func_text.append(f'joined_index = hpat_arrays_append(df_index, other_index)\n')
+        func_text.append(f'return pandas.DataFrame({{{data}}}, index=joined_index)\n')
```

Comment:
You could decrease number of words:
```suggestion
        func_text += [f'df_index = df.index',
                      f'other_index = other.index',
                      f'joined_index = hpat_arrays_append(df_index, other_index)\n',
                      f'return pandas.DataFrame({{{data}}}, index=joined_index)\n']
```
or
```suggestion
        func_text += [f'joined_index = hpat_arrays_append(df.index, other.index)\n',
                      f'return pandas.DataFrame({{{data}}}, index=joined_index)\n']
```